### PR TITLE
Add retry to cluster generation for testing in hack folder

### PR DESCRIPTION
**Pull request by todaywasawesome**
Signed-off-by: Dan Garfield <dan@codefresh.io>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [n/a] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

There are three very small changes to the resource generation
1. Switch to `helm upgrade --install` incase there is an attempt at repair, not really important, just stylistically better.
2. Add a few more debugging messages
3. Add a retry to getting cluster credentials. This step fails pretty regularly just because the endpoint isn't quite ready yet.

### CODE REVIEW

File: hack/gen-resources/generators/cluster_generator.go [hunk 1]
```diff
@@ -193,9 +195,23 @@ func (cg *ClusterGenerator) Generate(opts *util.GenerateOpts) error {
 
 		log.Print("Get cluster credentials")
 		caData, cert, key, err := cg.getClusterCredentials(namespace, releaseSuffix)
+
+		for o := 0; o < 5; o++ {
+			if err != nil {
+				if o > 0 {
```
Comment (crenshaw-dev) [inline hack/gen-resources/generators/cluster_generator.go:201]: I don't think we need the `if`... if we hit this line, we already know there was a failure, right?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: hack/gen-resources/generators/cluster_generator.go (hunk 1 above)
Comment (todaywasawesome) [reply hack/gen-resources/generators/cluster_generator.go]: @crenshaw-dev ah yes, you're right. I had that in there from a previous revision that I abandoned.

**Review by pasha-codefresh — APPROVED** [review]
LGTM

**Review by crenshaw-dev — APPROVED** [review]
lgtm!

**Pull request merged**